src/libraries/Common/src/Interop/Interop.Ldap.cs

```diff
@@ -0,0 +1,189 @@
+        LDAP_AUTH_SSPI = LDAP_AUTH_NEGOTIATE,
+        LDAP_AUTH_DIGEST = LDAP_AUTH_OTHERKIND | 0x4000,
+        LDAP_AUTH_EXTERNAL = LDAP_AUTH_OTHERKIND | 0x0020,
+        LDAP_AUTH_KRBV4 = 0xFF,
+        LDAP_AUTH_SIMPLE = 0x80
+    }
+
+    internal enum LdapOption
+    {
+        LDAP_OPT_DESC = 0x01,
+        LDAP_OPT_DEREF = 0x02,
+        LDAP_OPT_SIZELIMIT = 0x03,
+        LDAP_OPT_TIMELIMIT = 0x04,
+        LDAP_OPT_REFERRALS = 0x08,
+        LDAP_OPT_RESTART = 0x09,
+        LDAP_OPT_SSL = 0x0a, // Not Supported in Linux
+        LDAP_OPT_REFERRAL_HOP_LIMIT = 0x10, // Not Supported in Linux
+        LDAP_OPT_VERSION = 0x11,
+        LDAP_OPT_SERVER_CONTROLS = 0x12, // Not Supported in Windows
+        LDAP_OPT_CLIENT_CONTROLS = 0x13, // Not Supported in Windows
+        LDAP_OPT_API_FEATURE_INFO = 0x15,
+        LDAP_OPT_HOST_NAME = 0x30,
+        LDAP_OPT_ERROR_NUMBER = 0x31,
+        LDAP_OPT_ERROR_STRING = 0x32,
+        LDAP_OPT_SERVER_ERROR = 0x33,
+        LDAP_OPT_SERVER_EXT_ERROR = 0x34, // Not Supported in Linux
+        LDAP_OPT_HOST_REACHABLE = 0x3E, // Not Supported in Linux
+        LDAP_OPT_PING_KEEP_ALIVE = 0x36, // Not Supported in Linux
+        LDAP_OPT_PING_WAIT_TIME = 0x37, // Not Supported in Linux
+        LDAP_OPT_PING_LIMIT = 0x38, // Not Supported in Linux
+        LDAP_OPT_DNSDOMAIN_NAME = 0x3B, // Not Supported in Linux
+        LDAP_OPT_GETDSNAME_FLAGS = 0x3D, // Not Supported in Linux
+        LDAP_OPT_PROMPT_CREDENTIALS = 0x3F, // Not Supported in Linux
+        LDAP_OPT_TCP_KEEPALIVE = 0x40, // Not Supported in Linux
+        LDAP_OPT_FAST_CONCURRENT_BIND = 0x41, // Not Supported in Linux
+        LDAP_OPT_SEND_TIMEOUT = 0x42, // Not Supported in Linux
+        LDAP_OPT_REFERRAL_CALLBACK = 0x70, // Not Supported in Linux
+        LDAP_OPT_CLIENT_CERTIFICATE = 0x80, // Not Supported in Linux
+        LDAP_OPT_SERVER_CERTIFICATE = 0x81, // Not Supported in Linux
+        LDAP_OPT_AUTO_RECONNECT = 0x91, // Not Supported in Linux
+        LDAP_OPT_SSPI_FLAGS = 0x92,
+        LDAP_OPT_SSL_INFO = 0x93, // Not Supported in Linux
+        LDAP_OPT_SIGN = 0x95,
+        LDAP_OPT_ENCRYPT = 0x96,
+        LDAP_OPT_SASL_METHOD = 0x97,
+        LDAP_OPT_AREC_EXCLUSIVE = 0x98, // Not Supported in Linux
+        LDAP_OPT_SECURITY_CONTEXT = 0x99,
+        LDAP_OPT_ROOTDSE_CACHE = 0x9a // Not Supported in Linux
+    }
+
+    internal enum ResultAll
+    {
+        LDAP_MSG_ALL = 1,
+        LDAP_MSG_RECEIVED = 2,
+        LDAP_MSG_POLLINGALL = 3 // Not Supported in Linux
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal sealed class LDAP_TIMEVAL
+    {
+        public int tv_sec;
+        public int tv_usec;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal sealed class berval
+    {
+        public int bv_len = 0;
+        public IntPtr bv_val = IntPtr.Zero;
+
+        public berval() { }
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct SafeBerval
```

Comment:
It looks like this type can possibly be removed, it's the same as the previous one (`berval`), aside from class v struct.  (Would `berval` be better as a struct?)